static/js/sandbox-main.js

```diff
@@ -48,3 +48,138 @@ mobileNavTriggerButton.addEventListener('click', function () {
     mobileNavIcon.innerHTML = feather.icons['menu'].toSvg();
   }
 });
+
+// Sanbox Dashboard UI Application List Interactions
+// -------------------------------------------------
+
+// Dashboard App Credential Toggle
+// This shows/hides the client ID and secret for each application
+const appCredentialToggleLinks = document.querySelectorAll('.app-credentials-toggle');
+
+// Toggle on Mouse Click
+appCredentialToggleLinks.forEach((appCredentialToggleLink) => {
+  appCredentialToggleLink.addEventListener('click', function() {
+    let appID = document.querySelector('#id-' + this.id);
+    let appSecret = document.querySelector('#secret-' + this.id);
+    if (appID.type == 'password') {
+      appID.type = 'text';
+      appSecret.type = 'text';
+    } else {
+      appID.type = 'password';
+      appSecret.type = 'password';
+    }
+  });
+});
+
+// Toggle for Keyboard Naviagation
+appCredentialToggleLinks.forEach((appCredentialToggleLink) => {
+  appCredentialToggleLink.addEventListener('keypress', function (e) {
+    var key = e.which || e.keyCode;
+    if (key === 13) { // 13 is enter
+      let appID = document.querySelector('#id-' + this.id);
+      let appSecret = document.querySelector('#secret-' + this.id);
+      if (appID.type == 'password') {
+        appID.type = 'text';
+        appSecret.type = 'text';
+      } else {
+        appID.type = 'password';
+        appSecret.type = 'password';
+      }
+    }
+  });
+});
+
+// Copy Credentials to Clipboard
+function copyCredential(copyID) {
```

Comment:
The other behaviours in this file use unobtrusive javascript the reader can see both what the behaviour is and what is using it within this file. Since this is referenced directly from the HTML the reader needs more context to understand the usage of this behaviour. Just visually separating this function within the file (moving it to the bottom?) should be enough to help with the dissonance.

Reply:
Since we've cleaned up this logic a little bit, is this still bothering you as much?

Reply:
I'm thinking of a situation where a future developer is searching `js` files for usage of `copyCredential` and, not finding any, removing it without realizing it is in fact being used.

Reply:
Added comment to the code to make sure it is clear this this is being run from an HTML `onClick` situation. Fix has been added.